### PR TITLE
change snapshot controller image version to 6.0.1 in release-6.0 branch

### LINKED ISSUE #750: 6.0.1 release uses 5.0.1 image
file `./deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml` in 
https://github.com/kubernetes-csi/external-snapshotter/archive/refs/tags/v6.0.1.tar.gz

uses `gcr.io/k8s-staging-sig-storage/snapshot-controller:v5.0.1` image
should it be v6.0.1 ?

### PR DESCRIPTION
**Pull request by Jainbrt**
Signed-off-by: Abhishek Jain <abjain39@in.ibm.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug

> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #750

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
 
```release-note
None
```

### CODE REVIEW

**Comment by Jainbrt** [comment]
@xing-yang would you please help to review this PR ?

**Comment by Jainbrt** [comment]
Team, could you please help review the PR @andyzhangx @Jiawei0227 @xing-yang ?

**Review by andyzhangx — APPROVED** [review]
/lgtm

File: deploy/kubernetes/webhook-example/webhook.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
       serviceAccountName: snapshot-webhook
       containers:
       - name: snapshot-validation
-        image: k8s.gcr.io/sig-storage/snapshot-validation-webhook:v5.0.1 # change the image if you wish to use your own custom validation server image
+        image: k8s.gcr.io/sig-storage/snapshot-validation-webhook:v6.0.1 # change the image if you wish to use your own custom validation server image
```
Comment (xing-yang) [inline deploy/kubernetes/webhook-example/webhook.yaml:21]: Can you change this to registry.k8s.io/sig-storage/snapshot-validation-webhook:v6.0.1?

File: deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml [hunk 2]
```diff
@@ -33,7 +33,7 @@ spec:
       serviceAccountName: snapshot-controller
       containers:
         - name: snapshot-controller
-          image: gcr.io/k8s-staging-sig-storage/snapshot-controller:v5.0.1
+          image: gcr.io/k8s-staging-sig-storage/snapshot-controller:v6.0.1
```
Comment (xing-yang) [inline deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml:36]: registry.k8s.io/sig-storage/snapshot-controller:v6.0.1

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 3]
```diff
@@ -84,7 +84,7 @@ spec:
             - name: socket-dir
               mountPath: /csi
         - name: csi-snapshotter
-          image: gcr.io/k8s-staging-sig-storage/csi-snapshotter:v5.0.1
+          image: gcr.io/k8s-staging-sig-storage/csi-snapshotter:v6.0.1
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:87]: Change to registry.k8s.io/sig-storage/csi-snapshotter:v6.0.1

File: deploy/kubernetes/webhook-example/webhook.yaml (hunk 1 above)
Comment (Jainbrt) [reply deploy/kubernetes/webhook-example/webhook.yaml]: Done

**Pull request merged**